src/main/java/me/karakelley/tictactoe/Application.java

```diff
@@ -0,0 +1,29 @@
+package me.karakelley.tictactoe;
+
+public class Application {
+
+  private GameLoop gameLoop;
+  private Validator playAgainValidator;
+  private UserInterface userInterface;
+  private String PLAY_AGAIN = "Do you want to play again (Y / N)?\n";
+
+  public Application(UserInterface userInterface, GameLoop gameLoop, Validator playAgainValidator) {
+    this.userInterface = userInterface;
+    this.gameLoop = gameLoop;
+    this.playAgainValidator = playAgainValidator;
+  }
+
+  public void start(Game game, BoardState boardState) {
+    do gameLoop.start(game, boardState);
+    while (playAgain(boardState));
+  }
+
+  public boolean playAgain(BoardState boardState) {
+    String answer = userInterface.userPrompt(PLAY_AGAIN, playAgainValidator);
+    if (answer.toLowerCase().equals("y")) {
+      gameLoop.reset(boardState);
```

Comment:
`playAgain()` is supposed to return a boolean, but in the middle, it also has the side effect of resetting the board. It would make more sense for me if this only concerned itself with the decision to play the game or not. This class also has more knowledge than it needs of the gameLoop internals. It knows it has to reset a board and then start the game. Can it just call `repeatGame` or something similar, and the GameLoop will just know it has to reset the game as part of the repeat logic?